Source/WebCore/html/parser/HTMLParserIdioms.cpp

```diff
@@ -254,6 +254,44 @@ std::optional<double> parseValidHTMLFloatingPointNumber(StringView input)
     return parseValidHTMLFloatingPointNumberInternal(start, input.length());
 }
 
+template <typename CharacterType>
+static double parseHTMLFloatingPointNumberValueInternal(const CharacterType* position, size_t length, double fallbackValue)
+{
+    ASSERT(length > 0);
+
+    size_t leadingSpacesLength = 0;
+    while (leadingSpacesLength < length && isASCIIWhitespace(position[leadingSpacesLength]))
+        ++leadingSpacesLength;
+
+    position += leadingSpacesLength;
+    if (*position != '+' && *position != '-' && *position != '.' && !isASCIIDigit(*position))
+        return fallbackValue;
+
+    size_t parsedLength = 0;
+    double number = parseDouble(position, length - leadingSpacesLength, parsedLength);
+
+    // Numbers are considered finite IEEE 754 Double-precision floating point values.
+    ASSERT(-std::numeric_limits<double>::max() <= number || number < std::numeric_limits<double>::max());
+
+    // The following expression converts -0 to +0.
+    return number ? number : 0;
+}
+
+// https://html.spec.whatwg.org/#rules-for-parsing-floating-point-number-values
+double parseHTMLFloatingPointNumberValue(StringView input, double fallbackValue)
+{
+    if (input.isEmpty())
+        return fallbackValue;
```

Comment:
```cpp
    if (input.isEmpty())
        return fallbackValue;
```
> This probably isn’t needed if you deal with the issues above.

If we drop that from the code, we get failures for these tests:

https://github.com/WebKit/WebKit/blob/8639b7056d69938550ad662317bd8451b6765abf/LayoutTests/fast/dom/HTMLMeterElement/set-meter-properties.html#L12-L18

…because those need that `fallbackValue` in that _“before properties were set”_ case.

But maybe the failures are due to me not having sufficiently/correctly implemented all the code changes from your review comments?